Sources/Examples/EchoNIO/Generated/echo.grpc.swift

```diff
@@ -39,15 +39,18 @@ internal final class Echo_EchoService_NIOClient: GRPCServiceClient, Echo_EchoSer
   internal let connection: GRPCClientConnection
   internal var serviceName: String { return "echo.Echo" }
   internal var defaultCallOptions: CallOptions
+  internal var errorDelegate: ClientErrorDelegate?
```

Comment:
Should this be a weak reference?

Reply:
I don't _think_ so. We need someone to hold a strong reference to the delegate and the client seems like the obvious owner here since it's the "useful" object the client interacts with. If the user needs to reference the client in the delegate then they should use a weak reference in their delegate.

Reply:
"If the user needs to reference the client in the delegate then they should use a weak reference in their delegate." feels a bit like an inversion of responsibilities. But AFAICT the error delegate should exist before the client, anyway, so it can't really know about the client. So I guess this would be a non-issue in practice. It would also be a non-issue when using shared instances of the logger (or retaining it somewhere in "user land"), too.

Still not 100% sure about the right approach here, though.

Reply:
It would be possible to reference it in a delegate afterwards if the delegate stored a `Client?`, but I think that's a bit of an edge case and I'm not sure what we can do to prevent it (apart from documenting it). I'm also not 100% sure here but this the best candidate at the moment.